site/docs/v1/tech/oauth/integrations/cockroachdb.adoc

```diff
@@ -0,0 +1,61 @@
+
+After doing so, you'll be taken back to the [breadcrumb]#Authentication# page. Click [uielement]#Add# in [uielement]#Authentication Methods# to create a new provider, set [field]#Configuration# to `OIDC` and give it a meaningful name, like "FusionAuth", which will be displayed to your users when they browse to the login page. Click [uielement]#Submit# to go to the details page.
+
+In the method details page, click [uielement]#Edit# and fill in [field]#Issuer URL#, [field]#Client ID# and [field]#Client Secret# fields with the values you copied when creating the application in your FusionAuth instance.
+
+By default, someone will only be able to log in to your CockroachDB application if you have already created a user there with the same email address from your FusionAuth instance. However, you can enable CockroachDB to automatically create a user there with a `Developer` role if they don't exist by clicking [uielement]#Additional Settings# and setting the [uielement]#Auto-provision Accounts# switch to "Enabled". You can also define which email domains will be accepted by filling [field]#Allowed Email Domains#.
+
+image::oauth/cockroachdb/add-oidc.png[Adding FusionAuth OIDC provider to CockroachDB.,width=1200]
+
+Submit your changes by clicking [uielement]#Save# in the top right corner of the screen, which will redirect you back to the method details page, where you need to copy the [field]#Callback URL#. Go back to your FusionAuth instance and select [breadcrumb]#Applications#. Click the blue pencil icon to edit your application and paste that URL in [field]#Authorized redirect URLs# and save your application.
+
+Go back to CockroachDB and click the [uielement]#Test# button in [uielement]#Test Connection# to actually try out the whole process. If it succeeds, you can enable the integration by toggling the [field]#Status# switch.
+
+== Log in
+
+To test everything, open a different browser (or an incognito tab) and go to that custom URL you've configured earlier (something like `\https://cockroachlabs.cloud/login/your-url-here`). You'll now see a [uielement]#Log in with FusionAuth# (or whatever the name you chose) button. Click it and you will be prompted to log in to your FusionAuth instance. After completing the process, you should be redirected back to your CockroachDB DB Console already logged in.
+
+image::oauth/cockroachdb/login.png[Login screen for your CockroachDB instance.,width=1200]
+
+== Troubleshooting
+
+* Watch out when defining the issuer for your tenant and make sure you have included both the protocol (most commonly `https://`) and no trailing slashes.
+* Ensure that the signing keys for your FusionAuth application are using  asymmetric encryption (RS256) by navigating to [breadcrumb]#Settings -> Key Master#, locating the two keys named `Access token signing key generated for application My Next.js App` and `Id token signing key generated for application My Next.js App`. Both of them should have [uielement]#Type# as `RSA`.
```

Comment:
Yes, they are created exactly like that.

![image](https://user-images.githubusercontent.com/1877191/236950987-8ab8c887-a985-4bf5-b71c-15a5638858ba.png)

Reply:
Why is it a 'nextjs app' when it is for cockroachdb?

Reply:
Because one wouldn't normally name their app based on the DB they're using. But for the sake of the tutorial, I can rename it to `CockroachDB` if you want.

Reply:
Hmmm. I thought this was controlling access to the cockroachdb admin UI/console? In which case you would name the application for it.

Reply:
Yes, you're right. I've changed mentions to the old name and the screenshot.